internal/executor/job/job_run_state_store.go

```diff
@@ -127,7 +193,11 @@ func (stateStore *JobRunStateStore) Get(runId string) *RunState {
 	stateStore.lock.Lock()
 	defer stateStore.lock.Unlock()
 
-	return stateStore.jobRunState[runId].DeepCopy()
+	run, exists := stateStore.jobRunState[runId]
```

Comment:
Don't we need to mark the state as deleted (or actually delete it?). If we don't we should add a comment here as it's a bit weird to have a function called "Delete" that doesn't seem to delete anything!